### ISSUE TITLE
Implement automatic type casting for xs:untypedAtomic 

### ISSUE THREAD

**Issue by wschella** [comunica/sparqlee#14]
See [https://www.w3.org/TR/xpath-31/#dt-function-conversion](https://www.w3.org/TR/xpath-31/#dt-function-conversion). Relevant for all functions, somewhat important for unary plus.

**Comment by jitsedesmet** [comment]
I was wondering what this does? I don't really get it. :sweat_smile:

**Comment by wschella** [comment]
I think if things have a value of untypedAtomic we should cast them always? I don't understand very well either. Anyway it's not high priority.

**Comment by rubensworks** [comment]
Yeah, I've never seen untypedAtomic used anywhere so far, so I also don't think it's very urgent.

**Comment by jitsedesmet** [comment]
Okey, that mean this will not be handled in #102 like first proposed.
